### PR TITLE
ENT-3489: Save Nudge Emails Info in OfferAssignmentEmailSentRecord

### PR DESCRIPTION
**Pull request by sameenfatima78**
As the SRE team now has granted REFERENCES privileges on stage and prod db, we are finally good to merge the PRs again. This will undo the revert of ENT-3489 https://github.com/edx/ecommerce/pull/3246

### CODE REVIEW

File: ecommerce/enterprise/management/commands/send_code_assignment_nudge_emails.py [hunk 1]
```diff
@@ -35,6 +39,19 @@ def _get_nudge_emails():
             is_subscribed=True
         )
 
+    @staticmethod
+    def _create_email_sent_record(nudge_email):
+        """Creates an instance of OfferAssignmentEmailSentRecord with the given data."""
```
Comment (saleem-latif) [inline ecommerce/enterprise/management/commands/send_code_assignment_nudge_emails.py]: Please add argument details in the docstring, 
```
Creates an instance of OfferAssignmentEmailSentRecord with the given data.

Arguments:
    nudge_email (<instance-type>): <small-description>
```

File: ecommerce/enterprise/management/commands/send_code_assignment_nudge_emails.py [hunk 2]
```diff
@@ -35,6 +39,19 @@ def _get_nudge_emails():
...
+        code = nudge_email.code
+        user_email = nudge_email.user_email
+        enterprise_customer_uuid = get_enterprise_customer_uuid(code)
+        email_template = nudge_email.email_template
+        email_type = email_template.email_type
+        receiver_id = User.get_lms_user_id_from_email(user_email)
+        sender_category = AUTOMATIC_EMAIL
+        OfferAssignmentEmailSentRecord.create_email_record(enterprise_customer_uuid, email_type, email_template,
```
Comment (saleem-latif) [inline ecommerce/enterprise/management/commands/send_code_assignment_nudge_emails.py]: nit, I think the following would be more readable
```
OfferAssignmentEmailSentRecord.create_email_record(
    enterprise_customer_uuid=get_enterprise_customer_uuid(nudge_email.code), 
    email_type=email_template.email_type, 
    email_template=nudge_email.email_template, 
    sender_category=AUTOMATIC_EMAIL, 
    code=nudge_email.code, 
    user_email=nudge_email.user_email, 
    receiver_id=User.get_lms_user_id_from_email(user_email),
)
```

File: ecommerce/enterprise/management/commands/send_code_assignment_nudge_emails.py [hunk 3]
```diff
@@ -55,6 +72,7 @@ def handle(self, *args, **options):
                 send_nudge_email_count += 1
                 send_code_assignment_nudge_email.delay(nudge_email.user_email, email_subject, email_body)
                 self.set_last_reminder_date(nudge_email.user_email, nudge_email.code)
+                self._create_email_sent_record(nudge_email)
```
Comment (saleem-latif) [inline ecommerce/enterprise/management/commands/send_code_assignment_nudge_emails.py]: Should we add the record here or inside the task to make sure the record is added only when the email was sent successfully?

File: ecommerce/extensions/api/serializers.py [hunk 4]
```diff
@@ -1561,6 +1564,18 @@ def validate_order(self, order):
             raise serializers.ValidationError(_('Invalid order number or order {} does not exists.').format(order))
         return order
 
+    def _create_offer_assignment_email_sent_record(self, enterprise_customer_uuid, email_type, code, user_email):
+        """Records the new automated assignment email info sent after a successful refund."""
```
Comment (saleem-latif) [inline ecommerce/extensions/api/serializers.py]: Add Argument details in the doc string, like
```
Records the new automated assignment email info sent after a successful refund.

Arguments:
    enterprise_customer_uuid (<instance-type>): <small-description>
    email_type (<instance-type>): <small-description>
    code (<instance-type>): <small-description>
    user_email (<instance-type>): <small-description>
```

File: ecommerce/extensions/api/v2/views/enterprise.py [hunk 5]
```diff
@@ -671,6 +671,7 @@ def _validate_email_fields(self, subject, greeting, closing):
             raise DRFValidationError({'error': errors})
 
     def _create_offer_assignment_email_sent_record(self, enterprise_customer, email_type, template=None):
+        """Saves the all email info to OfferAssignmentEmailSentRecord."""
```
Comment (saleem-latif) [inline ecommerce/extensions/api/v2/views/enterprise.py]: Add information about the arguments, like
```
Arguments:
    enterprise_customer (<instance-type>): <small-description>
    email_type (<instance-type>): <small-description>
    template (<instance-type>): <small-description>
```

**Review by saleem-latif — APPROVED** [review]

File: ecommerce/enterprise/management/commands/send_code_assignment_nudge_emails.py (hunk 3 above)
Comment (sameenfatima78) [reply ecommerce/enterprise/management/commands/send_code_assignment_nudge_emails.py]: I think it makes more sense to add it here, we would still want to capture all the info of the email that was supposed to be sent. The table is supposed to capture the "usage" analytics of admin portal and nudge email features.

**Pull request closed without merging**